src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Engine/Trivia/CSharpTriviaFormatter.cs

```diff
@@ -72,14 +72,15 @@ protected override LineColumnRule GetLineColumnRuleBetween(SyntaxTrivia trivia1,
             // [trivia] [whitespace] [token] case
             if (trivia2.IsKind(SyntaxKind.None))
             {
-                var insertNewLine = this.FormattingRules.GetAdjustNewLinesOperation(this.Token1, this.Token2) != null;
```

Comment:
❔ Do you know why this is coming back as non-null?

Reply:
Seems we always generate it when we hit the EndOfLine token. Should that controlled by the `insert_final_newline` .editorconfig option?

https://github.com/dotnet/roslyn/blob/65d799af3a796c7928c5a6acfc961a15c9916d5f/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/EndOfFileTokenFormattingRule.cs#L15-L25

Reply:
I don't think we need to make a change to that code _yet_. Do we know why `IsStartOrEndOfFile` (a few lines above) isn't returning true for this case?

Reply:
https://github.com/dotnet/roslyn/blob/5b1cdefccd312a1b88e7e85a62d9ba918b87815e/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Engine/Trivia/CSharpTriviaFormatter.cs#L167-L168

Kinds:
trivia1 - MultiLineCommentTrivia (8542)
trivia2 - None (0)
Token1 - SemicolonToken (8212)
Token2 - EndOfFileToken (8496)

Reply:
@sharwell Updated IsStartOrEndOfFile to check for EndOfFileToken and reverted earlier changes.